packages/network-controller/tests/fake-provider.ts

```diff
@@ -106,21 +95,29 @@ export class FakeProviderEngine extends EventEmitter implements ProviderEngine {
    * of specific invocations of `sendAsync` matching a `method`.
    */
   constructor({ stubs = [] }: FakeProviderEngineOptions) {
-    super();
+    super({ engine: new JsonRpcEngine() });
     this.#originalStubs = stubs;
     this.#stubs = this.#originalStubs.slice();
     this.calledStubs = [];
-    this.#isStopped = false;
-    this._blockTracker = new FakeBlockTracker();
   }
 
-  stop() {
```

Comment:
Because you can't "stop" a SafeEventEmitter, we don't need to override `stop`, and we don't need to keep track of whether the provider is stopped.